homeassistant/components/sensor/forecast.py

```diff
@@ -48,21 +50,9 @@
 
 def setup_platform(hass, config, add_devices, discovery_info=None):
     """Setup the Forecast.io sensor."""
-    import forecastio
-
-    if None in (hass.config.latitude, hass.config.longitude):
-        _LOGGER.error("Latitude or longitude not set in Home Assistant config")
-        return False
-
-    try:
-        forecast = forecastio.load_forecast(config.get(CONF_API_KEY, None),
-                                            hass.config.latitude,
-                                            hass.config.longitude)
-        forecast.currently()
-    except ValueError:
-        _LOGGER.error(
-            "Connection error "
-            "Please check your settings for Forecast.io.")
+    api_key = config.get(CONF_API_KEY, None)
+    if None in (hass.config.latitude, hass.config.longitude, api_key):
+        _LOGGER.error("Latitude, longitude, or API key missing from config")
```

Comment:
I still think that the latitude/longitude error should be separate from the API key one. The settings for the latitude and longitude are configuration entries for HA itself while the API key is related to the configuration of the sensor thus they have different positions in the `configuration.yaml` file. This could mislead users.